scripts/core/editor3/reducers/spellchecker.jsx

```diff
@@ -1,25 +1,58 @@
 import {EditorState, Modifier} from 'draft-js';
+import ng from 'core/services/ng';
 
-/**
- * @ngdoc React
- * @module superdesk.core.editor3
- * @name Spellchecker Reducers
- * @description Contains the list of spellchecker related reducers.
- */
 const spellchecker = (state = {}, action) => {
     switch (action.type) {
     case 'SPELLCHECKER_REPLACE_WORD':
         return replaceWord(state, action.payload);
+    case 'SPELLCHECKER_IGNORE_WORD':
+        return ignoreWord(state, action.payload);
+    case 'SPELLCHECKER_ADD_WORD':
+        return addWord(state, action.payload);
     case 'SPELLCHECKER_SHOW_CONTEXT_MENU':
         return showContextMenu(state, action.payload);
+    case 'SPELLCHECKER_CLOSE_CONTEXT_MENU':
+        return closeContextMenu(state);
     default:
         return state;
     }
 };
 
 /**
  * @ngdoc method
- * @name Spellchecker Reducers#replaceWord
+ * @name ignoreWord
+ * @param {Object} state
+ * @param {String} word
+ * @return {Object} returns new state
+ * @description Replace the current word with the new selected one
+ */
+const ignoreWord = (state, text) => {
+    const spellcheck = ng.get('spellcheck');
+
+    spellcheck.addWord(text, true);
```

Comment:
Yes, this is still WIP. And yes, you're right, I would've missed this without your comment, so it's welcomed :)